r/R/dplyr.R

```diff
@@ -619,9 +619,23 @@ collect.arrow_dplyr_query <- function(x, as_data_frame = TRUE, ...) {
     restore_dplyr_features(tab, x)
   }
 }
-collect.ArrowTabular <- as.data.frame.ArrowTabular
+collect.ArrowTabular <- function(x, as_data_frame = TRUE, ...) {
+  if (as_data_frame) as.data.frame(x, ...) else x
+}
 collect.Dataset <- function(x, ...) dplyr::collect(arrow_dplyr_query(x), ...)
 
+compute.arrow_dplyr_query <- function(x, ...) dplyr::collect(x, as_data_frame = FALSE)
+compute.ArrowTabular <- function(x, ...) x
+compute.Dataset <- function(x, ...) {
+  dplyr::collect(arrow_dplyr_query(x), as_data_frame = FALSE, ...)
+}
+# for compatibility with dplyr 1.0.5 and earlier (https://github.com/tidyverse/dplyr/pull/5783):
+if ("name" %in% names(formals(dplyr::compute))) {
```

Comment:
(1) This doesn't seem necessary. The fact that you have `...` in the method signature will handle it fine. To confirm, I have dplyr with "name" in the signature locally, and I deleted this `if` block and the tests all pass. 

(2) Even if it were necessary, is this correct? This code is evaluated at install-time, so it's not robust to upgrading dplyr.

Reply:
This is to avoid the dreaded "S3 generic/method consistency" warnings from `R CMD check`, which by my recollection can be triggered if the method args do not include _all_ of the generic's formal args (dots or not).

Reply:
Right, but this technically isn't an S3 method as far as `R CMD check` is concerned because we register it onLoad only. It's not listed as an S3 method in `NAMESPACE`

Reply:
Oh, gotcha. Then it should be safe to remove this 👍

Reply:
Removed in ec744ecb3dd788051873002784fa061d4ae19223